contract_api/registry.py

```diff
@@ -0,0 +1,274 @@
+                all_orgs_members[rec['org_id']].append(rec['member'])
+            return all_orgs_members
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def get_all_org(self):
+        """ Method to get list and high level details of all organization."""
+        try:
+            all_orgs_srvcs = self._get_all_service()
+            all_orgs_members = self._get_all_members()
+            all_orgs_data = []
+            for rec in all_orgs_srvcs:
+                data = {"org_id": rec,
+                        "owner_address": all_orgs_srvcs[rec]['owner_address'],
+                        "service_id": all_orgs_srvcs[rec]['service_id'],
+                        "members": all_orgs_members.get(rec, [])}
+                all_orgs_data.append(data)
+            return all_orgs_data
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _prepare_subquery(self, s, q, fm):
+        try:
+            sub_qry = ""
+            if s == "all":
+                for rec in fm:
+                    sub_qry += fm[rec] + " LIKE '" + str(q) + "%' OR "
+                sub_qry = sub_qry[:-3] if sub_qry.endswith("OR ") else sub_qry
+            else:
+                sub_qry += fm[s] + " LIKE '" + str(q) + "%' "
+            return sub_qry.replace("org_id", "M.org_id")
+        except Exception as err:
+            raise err
+
+    def _get_total_count(self, sub_qry, filter_query, values):
+        try:
+            if filter_query != "":
+                filter_query = " AND " + filter_query
+            search_count_query = "SELECT count(*) as search_count FROM service A, (SELECT DISTINCT M.org_id, M.service_id FROM " \
+                                 "service_metadata M LEFT JOIN service_tags T ON M.service_row_id = T.service_row_id WHERE (" \
+                                 + sub_qry.replace('%', '%%') + ")" + filter_query + ") B WHERE A.service_id = B.service_id " \
+                                 "AND A.org_id = B.org_id AND A.is_curated = 1 "
+
+            res = self.repo.execute(search_count_query, values)
+            return res[0].get("search_count", 0)
+        except Exception as err:
+            raise err
+
+    def _search_query_data(self, sub_qry, sort_by, order_by, offset, limit, filter_query, values):
+        try:
+            if filter_query != "":
+                filter_query = " AND " + filter_query
+            srch_qry = "SELECT * FROM service A, (SELECT M.org_id, M.service_id, group_concat(T.tag_name) AS tags FROM " \
+                       "service_metadata M LEFT JOIN service_tags T ON M.service_row_id = T.service_row_id WHERE (" \
+                       + sub_qry.replace('%', '%%') + ")" + filter_query + " GROUP BY M.org_id, M.service_id ORDER BY %s %s ) B WHERE " \
+                                                      "A.service_id = B.service_id AND A.org_id=B.org_id AND A.is_curated= 1 LIMIT %s , %s"
```

Comment:
We must enforce some thresholds for offset and limit to ensure that bound the data returned. For example offset 0 and limit 10000 may end up pulling 10000 rows. We probably need to enforce this higher up in the flow rather than at this point